klippy/gcode.py

```diff
@@ -349,6 +360,24 @@ def _process_data(self, eventtime):
         if self.fd_handle is None:
             self.fd_handle = self.reactor.register_fd(self.fd,
                                                       self._process_data)
+    def _handle_remote_help(self, web_request):
+        web_request.send(dict(self.gcode_help))
+    def _handle_remote_restart(self, web_request):
+        path = web_request.get_path()
+        if path == "gcode/restart":
+            web_request.put('script', 'restart')
+        elif path == "gcode/firmware_restart":
+            web_request.put('script', 'firmware_restart')
+        self.run_script_from_remote(web_request)
+    def run_script_from_remote(self, web_request):
+        script = web_request.get('script')
+        if 'M112' in script.upper():
+            self.cmd_M112({})
+            return
+        try:
+            self.run_script(script)
+        except Exception as e:
+            raise web_request.error(e.message, 400)
```

Comment:
Mentioned in a previous review comment, but I don't think it would be safe to catch all errors.  If the webhook dispatch code is modified to catch CommandError() then this error handler could probably be removed though.

Also, separately, I wonder if we need to specify the http error code in the klippy api.  As mentioned in a previous comment, if the api is thought of more as a "json api" instead of an "http api" it might be more applicable to other users.

Reply:
Agreed, I will make the suggested modifications to webhooks and remove exception handling here.

With regard to the status code, it can be helpful for web clients to receive a status code when they are doing their own error handling.  However we could abstract the status code from Klippy with our own list of errors.  For example, instead of sending "400", we could simply send "Bad Request" or something more applicable to Klippy.  The server could then look up the appropriate status code and return it to the client upon an error.

Reply:
Okay.  One simple solution is to always use error code 400 for bad api requests.

Reply:
I can do that.   The status code is accompanied by the error message, so client devs should be able work with it.